tests/test_job_manager.py

```diff
@@ -10,45 +10,64 @@
         docker_img=expected_image, cmd=expected_command,
         env_vars={expected_env_var_name: expected_env_var_value},
         workflow_uuid=workflow_uuid)
     with mock.patch("reana_job_controller.kubernetes_job_manager."
                     "current_k8s_batchv1_api_client") as kubernetes_client:
-        with patch.multiple("reana_job_controller.job_manager",
-                            Session=DEFAULT,
-                            calculate_file_access_time=DEFAULT) as db_mock:
-            job_id = job_manager.execute()
-            kubernetes_client.create_namespaced_job.assert_called_once()
-            body = kubernetes_client.create_namespaced_job.call_args[1]['body']
-            env_vars = body['spec']['template']['spec']['containers'][0]['env']
-            image = body['spec']['template']['spec']['containers'][0]['image']
-            command = \
-                body['spec']['template']['spec']['containers'][0]['command']
-            assert len(env_vars) == 1
-            assert env_vars[0]['name'] == expected_env_var_name
-            assert env_vars[0]['value'] == expected_env_var_value
-            assert image == expected_image
-            assert command == expected_command
+        kubernetes_job_id = job_manager.execute()
+        kubernetes_client.create_namespaced_job.assert_called_once()
+        body = kubernetes_client.create_namespaced_job.call_args[1]['body']
+        env_vars = body['spec']['template']['spec']['containers'][0]['env']
+        image = body['spec']['template']['spec']['containers'][0]['image']
+        command = \
+            body['spec']['template']['spec']['containers'][0]['command']
+        assert len(env_vars) == 1
+        assert env_vars[0]['name'] == expected_env_var_name
+        assert env_vars[0]['value'] == expected_env_var_value
+        assert image == expected_image
+        assert command == expected_command
 
 
-def test_stop_kubernetes_job():
+def test_stop_kubernetes_job(app, session, sample_serial_workflow_in_db,
+                             sample_workflow_workspace):
     """Test stop of Kubernetes job."""
-    pass
+    workflow_uuid = sample_serial_workflow_in_db.id_
+    next(sample_workflow_workspace(
+        str(workflow_uuid)))
+    expected_env_var_name = "env_var"
+    expected_env_var_value = "value"
+    expected_image = "busybox"
+    expected_command = ["ls"]
+    job_manager = KubernetesJobManager(
+        docker_img=expected_image, cmd=expected_command,
+        env_vars={expected_env_var_name: expected_env_var_value},
+        workflow_uuid=workflow_uuid)
+    with mock.patch("reana_job_controller.kubernetes_job_manager."
+                    "current_k8s_batchv1_api_client") as kubernetes_client:
+        kubernetes_job_id = job_manager.execute()
+        kubernetes_client.create_namespaced_job.assert_called_once()
+        job_manager.stop(kubernetes_job_id)
+        kubernetes_client.delete_namespaced_job.assert_called_once()
```

Comment:
Shall we, for both tests, make sure that DB is modified in the correct way? since we are setting DB write/reads as responsibility for this class I think we should test it.